kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/dsl/FromFileCreatable.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.dsl;
+
+public interface FromFileCreatable<T> {
```

Comment:
Do you anticipate using this across types?  If not, it may be best to merge with ConfigMapResource or wherever this functionality lands.  Templates have a similar bit of functionality, but it's based upon load as it does not support directories.

Reply:
I thought this might be useful in case we decide to support DSL method for creating Secrets from file in future. There is a `--from-file` option in `kubectl` for Secrets as well. Do you think this makes sense? Otherwise I can merge this with ConfigMapResource.

Reply:
> Do you think this makes sense?

That make sense, just make sure to remove the reference to ConfigMap.  Based upon kubectl this could even be a vararg of dirOrFilePath.